lib/topology_openswitch/vtysh.py

```diff
@@ -170,8 +171,27 @@ def _determine_set_prompt(self, connection=None):
         # always return the produced output, even if an EOF exception
         # follows after it. This is done to handle the segmentation fault
         # errors.
-        spawn.sendline('stdbuf -oL vtysh')
-        spawn.expect(VTYSH_STANDARD_PROMPT)
+        for i in range(10):
+                try:
+                    spawn.sendline('stdbuf -oL vtysh')
+                    spawn.expect(VTYSH_STANDARD_PROMPT, timeout=3)
+                    break
+                except TIMEOUT:
+                    continue
+                except Exception as error:
+                    warning(
+                        'Unable to connect due  to unkown error: {}'.format(
+                            str(error)
+                        )
+                    )
+
+        else:
+            # Decode what is found to uft-8 and prints
+            before_msg = spawn.before.decode('utf-8', errors='ignore')
+            raise Exception(
+              'Unable to connect to vytsh, before: {}'.format(
```

Comment:
You know what ``before`` means but the user that may read this error will not necessarily know that. Change it to something easier to understand, like "the last output received was".